module/zfs/spa.c

```diff
@@ -10229,6 +10343,7 @@ spa_sync(spa_t *spa, uint64_t txg)
 	}
 
 	spa_sync_rewrite_vdev_config(spa, tx);
+	spa_sync_time_logger(spa, tx);
```

Comment:
It seems you are dirtying the pool too late in a sync process. It does not need to be that late. You could move it earlier, somewhere around `brt_pending_apply()`.